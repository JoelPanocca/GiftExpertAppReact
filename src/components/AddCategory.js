import React, { useState } from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ( {setCategories} ) => {
    const [newCategory, setNewCategory] = useState('')
    const handleOnSubmit = (e) => {
        e.preventDefault();
        if (newCategory.trim().length > 2) {
            setCategories( cats => [newCategory, ...cats]);
            setNewCategory('');
        }
        
    }

    const handleOnChange = (e) => {
        setNewCategory(e.target.value);
    }
    return (
        <form onSubmit={handleOnSubmit}>
            <input type = "text" value={newCategory} onChange={handleOnChange} placeholder="Escriba mÃ¡s de 3 letras y enter"/> 
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
