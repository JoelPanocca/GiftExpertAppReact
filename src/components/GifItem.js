import React from "react";
import PropTypes from 'prop-types';

export const GifItem = (img) => {
  return (
    <div className="card">
      <p>{img.titulo}</p>
      <img src={img.url} alt={img.titulo} />
    </div>
  );
};

GifItem.propTypes = {
    titulo : PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
