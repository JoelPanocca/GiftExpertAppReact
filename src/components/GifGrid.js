import React from "react";
import { useFetchGif } from "../hooks/useFetchGif";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {
  const {data:images, loading } = useFetchGif(category);
  return (
    <>
      <h2>{category}</h2>
      {loading && <p>Cargando...</p>}

      <div className="card-grid">
        {images.map((image) => {
          return <GifItem key={image.id} {...image} />;
        })}
      </div>
    </>
  );
};
