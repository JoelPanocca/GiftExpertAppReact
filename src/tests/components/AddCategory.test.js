import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe("Pruebas en el componente <AddCategory />", () => {
  const setCategoriesFun = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategoriesFun} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategoriesFun} />);
  });

  test("El componente debe mostrarse segun su snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de cambiar el valor de la caja de texto", () => {
    const input = wrapper.find("input");
    const valor = "Pokemon";
    input.simulate("change", { target: { value: valor } });
    expect(wrapper.find("p").text().trim()).toBe(valor);
  });

  test("debe de llamar a setCategories y limpiar la caja de texto", () => {
    const valor = "Hola Mundo";
    //1. Simular el inputChange
    wrapper.find("input").simulate("change", { target: { value: valor } });

    //2. Simular el submit
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    //3. SetCategories se debe haber llamado
    expect(setCategoriesFun).toHaveBeenCalled();
    expect(setCategoriesFun).toHaveBeenCalledTimes(1);

    //4. El valor del input debe ser ''
    expect(wrapper.find("input").prop("value")).toBe("");
  });
});
