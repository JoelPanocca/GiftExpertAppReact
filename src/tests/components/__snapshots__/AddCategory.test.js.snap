// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategory /> El componente debe mostrarse segun su snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    placeholder="Escriba mÃ¡s de 3 letras y enter"
    type="text"
    value=""
  />
</form>
`;
